Sources/POSIX/exec.swift

```diff
@@ -0,0 +1,19 @@
+/*
+ This source file is part of the Swift.org open source project
+ 
+ Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import libc
+
+/// Replace the current process image with a new process image.
+public func exec(_ path: String, _ args: [String]) throws {
+    let nullTerminatedArgs = args.map({ $0.withCString(strdup) }) + [nil]
```

Comment:
We will need to free this. We use similar pattern in many places now, we should get a utility method for this.